resteasy-core/src/main/java/org/jboss/resteasy/core/MediaTypeMap.java

```diff
@@ -550,8 +551,8 @@ private void mergeEverything(Entry<T> entry) {
       CachedMediaTypeAndClass cacheEntry = null;
       if (useCache)
       {
+         cacheEntry = new CachedMediaTypeAndClass(type, accept);

```

Comment:
Why does this line move out of check block ?

Reply:
This is just because the first time when the classCache is null the cacheEntry is not initialized, it remains null, and therefore the put at the end is done with cacheEntry=null, so nothing is cached the first time (indeed it's cached wrong, with a wrong key). I just fixed this minor issue too. This change is minor, as this is just executed differently the first time when classCache is null, all the rest of times is exactly the same.